learningpath-api/src/main/scala/no/ndla/learningpathapi/service/UpdateService.scala

```diff
@@ -420,6 +424,128 @@ trait UpdateService {
         case t: Throwable => throw t
       }
     }
+
+    def newFolder(
+        newFolder: api.NewFolder,
+        feideAccessToken: Option[FeideAccessToken] = None
+    ): Try[api.Folder] = {
+      for {
+        feideId         <- feideApiClient.getUserFeideID(feideAccessToken)
+        newDomainFolder <- folderRepository.insertFolder(converterService.toDomainFolder(newFolder, feideId))
+        api             <- converterService.toApiFolder(newDomainFolder)
+      } yield api
+    }
+
+    def newFolderResourceConnection(
+        folderId: Long,
+        newResource: api.NewResource,
+        feideAccessToken: Option[FeideAccessToken] = None
+    ): Try[_] = {
+      for {
+        feideId <- feideApiClient.getUserFeideID(feideAccessToken)
+        _ <- folderRepository
+          .folderWithFeideId(folderId, feideId)
+          .orElse(Failure(NotFoundException(s"Can't connect resource to non-existing folder")))
+        resource <- getExistingResourceOrCreateNew(newResource, feideId)
+        x        <- resource.doIfIdExists(id => folderRepository.createFolderResourceConnection(folderId, id))
+      } yield x
+    }
+
+    private[service] def getExistingResourceOrCreateNew(
+        newResource: api.NewResource,
+        feideId: FeideID
+    ): Try[domain.Resource] = {
+      folderRepository
+        .resourceWithPathAndFeideId(newResource.path, feideId)
+        .flatMap(
+          {
+            case None =>
+              val converted = converterService.toDomainResource(newResource, feideId)
+              folderRepository.insertResource(converted)
+            case Some(existingResource) =>
+              Success(converterService.mergeResource(existingResource, newResource))
+          }
+        )
```

Comment:
Er ikke så farlig, men jeg liker bedre å ikke ha med de ekstra parantesene :^)
```suggestion
        .flatMap {
          case None =>
            val converted = converterService.toDomainResource(newResource, feideId)
            folderRepository.insertResource(converted)
          case Some(existingResource) =>
            Success(converterService.mergeResource(existingResource, newResource))
        }
```